components/mdns/mdns.c

```diff
@@ -3492,7 +3492,7 @@ void mdns_parse_packet(mdns_rx_packet_t *packet)
     mdns_search_once_t *search_result = NULL;
 
 #ifdef MDNS_ENABLE_DEBUG
-    _mdns_dbg_printf("\nRX[%u][%u]: ", packet->tcpip_if, (uint32_t)packet->ip_protocol);
+    _mdns_dbg_printf("\nRX[%zu][%u]: ", packet->tcpip_if, (uint32_t)packet->ip_protocol);
```

Comment:
suggest casting the variable instead of using `%zu` format, since it's not supported in nano-format lib.